src/test/java/com/myFirstProject/myFirstProject/service/AccountServiceImplTest.java

```diff
@@ -0,0 +1,325 @@
+        Assert.assertNotNull(payment);
+        Assert.assertNotNull(payment.getTimeOfPayment());
+        Assert.assertEquals(BigDecimal.TEN, payment.getSum());
+        Assert.assertEquals(buildCurrencyEntity(), payment.getCurrencyEntity());
+        Assert.assertEquals(BigDecimal.ONE, payment.getRate());
+        Assert.assertNotNull(payment.getAccount());
+
+        Mockito.verify(accountRepository).save(eq(buildNewAccount()));
+
+    }
+
+    @Test(expected = UserNotFoundException.class)
+    public void updateAccountWithException() {
+//        GIVEN
+        AccountServiceImpl accountService = new AccountServiceImpl();
+        accountService.setUserRepository(userRepository);
+        Mockito.when(userRepository.findById(1l)).thenReturn(Optional.empty());
+//        WHEN
+        accountService.updateAccount(buildPaymentReq());
+    }
+
+    @Test
+    public void currencyConverter() {
+        //Given
+        BigDecimal sum = BigDecimal.TEN;
+        CurrencyEntity givenCurrency = new CurrencyEntity();
+        givenCurrency.setCurrency(Currency.EUR);
+        AccountServiceImpl accountService = new AccountServiceImpl();
+        accountService.setCurrencyRateService(currencyRateService);
+        Mockito.when(currencyRateService.getRateFromRepository()).thenReturn(buildCurrencyRates());
+//        When
+        BigDecimal actualResult = accountService.currencyConverter(sum, givenCurrency);
+//        Then
+        Assert.assertNotNull(actualResult);
+        Assert.assertEquals(BigDecimal.valueOf(5.0), actualResult);
+
+    }
+
+    @Test
+    public void calculateTaxPerMonth() {
+        //Given
+        List<Payment> payments = buildPaymentList();
+        AccountServiceImpl accountService = new AccountServiceImpl();
+        accountService.setCurrencyRateService(currencyRateService);
+        Mockito.when(currencyRateService.getRateFromRepository()).thenReturn(buildCurrencyRates());
+        //WHEN
+        BigDecimal actualResult = accountService.calculateTaxPerMonth(payments);
+// Then
+        Assert.assertNotNull(actualResult);
+        Assert.assertEquals(BigDecimal.valueOf(0.5), actualResult);
+    }
+
+    private List<CurrencyRate> buildCurrencyRates() {
+        CurrencyRate currencyRate = new CurrencyRate();
+        CurrencyEntity original = new CurrencyEntity();
+        original.setCurrency(Currency.USD);
+        CurrencyEntity destination = new CurrencyEntity();
+        destination.setCurrency(Currency.EUR);
+        currencyRate.getCompositeId().setOriginal(original);
+        currencyRate.getCompositeId().setDestination(destination);
+        currencyRate.setRate(BigDecimal.valueOf(0.5));
+        List<CurrencyRate> currencyRates = new ArrayList<>();
+        currencyRates.add(currencyRate);
+
+        return currencyRates;
+    }
+
+    private List<Payment> buildPaymentList() {
+        List<Payment> paymentList = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
```

Comment:
it is bad style to use a cycle for test objects.
It should be new method 
buildPayment(Currency currency, ...)

Reply:
dont understant your comment

Reply:
right code:
private List<Payment> buildPaymentList() {
List<Payment> paymentList = new ArrayList<>();
paymentList.add(buildPayment(1))
paymentList.add(buildPayment(2))
paymentList.add(buildPayment(4))
}

Reply:
but must feel with field by currency entity and sum

Reply:
buildPayment(sum, currency)